### PR TITLE
Adding fix for the site service

### PR DESCRIPTION
**Pull request by cody-constine-ttd**

### CODE REVIEW

**Review by ssundahlTTD — APPROVED** [review]

File: src/main/java/com/uid2/admin/vertx/service/SiteService.java [hunk 1]
```diff
@@ -99,9 +101,10 @@ private void handleSiteList(RoutingContext rc) {
             final Collection<Site> sites = this.siteProvider.getAllSites().stream()
                     .sorted(Comparator.comparing(Site::getName))
                     .collect(Collectors.toList());
-            final Map<Integer, List<ClientKey>> clientKeys = this.clientKeyProvider.getAll().stream()
-                    .collect(Collectors.groupingBy(ClientKey::getSiteId));
+            final Map<Integer, List<LegacyClientKey>> legacyClientKeys = this.legacyClientKeyProvider.getAll().stream()
+                    .collect(Collectors.groupingBy(LegacyClientKey::getSiteId));
             final List<ClientKey> emptySiteKeys = new ArrayList<>();
+            final List<LegacyClientKey> emptySiteLegacyKeys = new ArrayList<>();
```
Comment (asloobq) [inline src/main/java/com/uid2/admin/vertx/service/SiteService.java:107]: `emptySiteKeys` is unused

File: src/test/java/com/uid2/admin/vertx/SiteServiceTest.java [hunk 2]
```diff
@@ -26,7 +26,7 @@
 public class SiteServiceTest extends ServiceTestBase {
     @Override
     protected IService createService() {
-        return new SiteService(auth, writeLock, storeWriter, siteProvider, clientKeyProvider);
+        return new SiteService(auth, writeLock, storeWriter, siteProvider, legacyClientKeyProvider);
```
Comment (asloobq) [inline src/test/java/com/uid2/admin/vertx/SiteServiceTest.java:29]: I think you can update `ServiceTestBase.setClientKeys()` to remove references to `clientKeyProvider` and make it unambigous that only LegacyClientKeyProvider is used

**Review by asloobq — APPROVED** [review]
Added some minor comments which are good to be addressed later. Not needed if this is  an urgent fix.

File: src/test/java/com/uid2/admin/vertx/SiteServiceTest.java (hunk 2 above)
Comment (gmsdelmundo) [reply src/test/java/com/uid2/admin/vertx/SiteServiceTest.java:29]: Let me do this in my own PR

**Review by gmsdelmundo — APPROVED** [review]

**Pull request merged**